### DISCUSSION TITLE
Different outputs for the same transform

### DISCUSSION THREAD

**Discussion by Sebagam** [Project-MONAI/MONAI#3663] [Q&A]
Hi all,

I'm using two different nifti file datasets, I apply the same transform to them:

Compose([AsChannelFirst(), Resize(spatial_size = [250, 250, 170]), ScaleIntensity(), EnsureType()])

For one group of files I get this tensor:
<class 'torch.Tensor'> torch.Size([2, 1, 250, 250, 170]) tensor([1, 1])

While for the other, I get:
<class 'torch.Tensor'> torch.Size([1, 160, 250, 250, 170]) tensor([1])

What is the issue with the output from the second group and how I can fix it?

Thanks

**Comment by Nic-Ma** [comment]
Hi @Sebagam ,

Could you please help share your full test program for further analysis?
I feel the data shape you printed out is not expected as channel-first.

Thanks.

**Comment by rijobro** [comment]
The input to our pre-processing transforms is `C, H, W, [D]`. It looks like your images do not conform to this, as your first example has a leading `[2, 1]` and then three spatial dimensions. Your second example has a leading `[1, 160]`, followed by three spatial dimensions.

`Resize` modifies the images spatial dimensions. You've requested output spatial dimensions of `250, 250, 170`. Looking at the output shapes of your examples, they both match in the spatial dimensions.

**Comment by Sebagam** [comment]
It's [batch size, channels, dim1, dim2, dim3], right?
Do you know why I'm getting 175 channels in the test datasets and how to fix it?
Thanks

**Comment by rijobro** [comment]
Nifti images don't have batch components, they have the shape `x, y, z, t, u, v, w`. So I'm not sure what your 2 refers to in your second example. Either way, MONAI doesn't support the batch dimension, and will have to be removed (you could split them into separate images). I would recommend looping over **all** of your images with something like:

```python
loader = LoadImage()
for filename in list_of_filenames:  # train, validation and test data
    im, meta = loader(filename)
    print(meta["dim"])
```
It's important to know the variation in shape of your data to choose your transforms accordingly.

c.f.: info on nifti dimensions: https://nifti.nimh.nih.gov/nifti-1/documentation/nifti1fields/nifti1fields_pages/dim.html/document_view
